### ISSUE TITLE
Dates before the year 1900 can't be displayed on frontend

### ISSUE THREAD

**Issue by jansier** [odoo/odoo#41788]
Impacted versions: 13 CE

Steps to reproduce:
Save a date before 1900-01-01 on a date field using backend.
Example using odoo shell:
inv = env['account.move'].browse([1])
inv.write({'invoice_date': '1850-01-01'})
self.env.cr.commit()
When the date is saved, try to open form view with the date field of the given record.

Current behavior:
The date is saved using backend, but when the date field should be displayed on frontend exception is raised. "Error: '1850-01-01' is not a correct date"

Expected behavior:
Either make it possible to display old dates on frontend, or constrain the year > 1900 on backend.

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/25088953/70712782-d92dcc80-1ce4-11ea-8625-81fca722a5a5.png)

**Comment by pedrobaeza** [comment]
That's because you are hacking this, as that date is not allowed to be saved via the web client. The limitation is in JS itself.

I would close this as invalid, but I let @Yenthe666 to give a second opinion.

**Comment by jansier** [comment]
Ok, I gave probably a bad example then if it seems to be hacking. A proper example would be to import such a date from a .csv file. Web client allows such an import.

**Comment by Yenthe666** [comment]
@pedrobaeza is in fact right, it's simply a limitation of the framework right now. There have been quite some bug reports about this in the past. They where all closed with pretty much the same response. It's not supported, Odoo doesn't want to modify it so if you really need it you'll need to handle the case yourself. Sorry.

**Comment by jansier** [comment]
I see. If you say so, then ok. Just the last thing that I want to say @Yenthe666  is that the outcome of such a situation can be actually really tremendous. For me this error occurred in Invoicing app, that is used in our Odoo instance by over 100 hundred users. Imagine that a user with just create access right can, by a simple mistake, import an invoice with a wrong date. What are the results? Nobody, not even administrator can access List view of invoices, nobody can fix this date because web client gets crazy. The only way to handle this wrong date is to fix it directly in the database..

**Comment by Yenthe666** [comment]
Yeah I agree with you that it is far from ideal. Perhaps @mart-e can have another look at this one but I assume the opinion hasn't changed though. @mart-e perhaps we could at least raise a validation error or something?

**Comment by mart-e** [comment]
If it is a limitation of the datepicker, it should probably be reported upstream, at @tempusdominus

**Comment by mart-e** [comment]
After a bit more research, as discussed with @aab-odoo there is a check in our code stating all dates must be over 1900, the issue may not be in the datepicker, sorry.

https://github.com/odoo/odoo/blob/157e3fe4d4fb47b33211d1e52116284090b3e01a/addons/web/static/src/js/fields/field_utils.js#L435-L442

https://github.com/odoo/odoo/blob/157e3fe4d4fb47b33211d1e52116284090b3e01a/addons/web/static/src/js/widgets/date_picker.js#L33

I am not sure why. It seems that timezone were not properly standardized before 1900 which could explain the default limit.

**Comment by Yenthe666** [comment]
@mart-e just two ideas from the top of my head:
1) Allow dates to be before 1900 and document it as a limitation that timezones go wrong
2) Make it a configurable option to toggle "Dates before 1900" on or off

I've seen the issue a few times now so I guess it has to bother more people.

**Comment by mart-e** [comment]
After a bit more research with the help of @xmo-odoo, we found out before python 3.2, we were not able to add dates prior to 1900. In python 3.7, we can go all the way to year 1.
I don't have any issue to remove this two hardcoded limitations. As it may have other strange side effects, I prefer if we do it in master though.

**Comment by Yenthe666** [comment]
Let's reopen it to keep track of this then! Thanks for digging further into this guys. @mart-e can you create a quick PR or do you want me to open an OPW?

**Comment by mart-e** [comment]
> can you create a quick PR or do you want me to open an OPW?

Sorry but even if the limitation is no longer relevant (it was before v11), it is still a wishlist so neither. In case somebody wants to make a PR to remove the limitation, feel free to do so and the web framework team that will review it (not my area of expertise).

**Comment by C3POdoo** [comment]
Dear @jsierp,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
